tools/test/export/build_test.py

```diff
@@ -17,165 +17,257 @@
+        try:
+            exporter = export(test_case.mcu, test_case.ide,
+                              project_id=test_case.id, zip_proj=None,
+                              clean=True, src=test_case.src,
+                              export_path=path.join(EXPORT_DIR,name_str),
+                              silent=True)
+            self.build_queue.put((exporter,test_case))
+        except TargetNotSupportedException:
+            self.skips.append("%s::%s\t%s" % (test_case.mcu, test_case.ide,
+                                              test_case.name))
+            # Check if the specified name is in all_os_tests
 
-        """
-        successes = []
-        failures = []
-        skips = []
-        for mcu, ide in self.mcu_ide_pairs:
-            for test in self.tests:
-                export_location, name, src, lib = setup_project(ide, mcu,
-                                                                program=test)
-                test_name = Test(test).id
-                try:
-                    exporter = export_project(src, export_location, mcu, ide,
-                                              clean=clean, name=name,
-                                              libraries_paths=lib)
-                    exporter.progen_build()
-                    successes.append("%s::%s\t%s" % (mcu, ide, test_name))
-                except FailedBuildException:
-                    failures.append("%s::%s\t%s" % (mcu, ide, test_name))
-                except TargetNotSupportedException:
-                    skips.append("%s::%s\t%s" % (mcu, ide, test_name))
-
-                ProgenBuildTest.handle_log_files(export_location, ide, name)
-                if clean:
-                    shutil.rmtree(export_location, ignore_errors=True)
-        return successes, failures, skips
 
+def check_valid_mbed_os(test):
+    """Check if the specified name is in all_os_tests
+    args:
+        test: string name to index all_os_tests
+    returns: tuple of test_name and source location of test,
+        as given by find_tests"""
+    all_os_tests = find_tests(ROOT, "K64F", "ARM")
+    if test in all_os_tests.keys():
+        return (test, all_os_tests[test])
+    else:
+        supported = columnate([t for t in all_os_tests.keys()])
+        raise ArgumentTypeError("Program with name '{0}' not found. "
+                                "Supported tests are: \n{1}".format(test,supported))
+def check_version(version):
+    """Check if the specified version is valid
+    args:
+        version: integer versio of mbed
+    returns:
+        version if it is valid"""
+    if version not in RELEASE_VERSIONS:
+        raise ArgumentTypeError("Choose from versions : %s"%", ".join(RELEASE_VERSIONS))
+    return version
 
 def main():
     """Entry point"""
-    toolchainlist = ["iar", "uvision", "uvision5"]
-    default_tests = [test_name_known("MBED_BLINKY")]
-    targetnames = TARGET_NAMES
-    targetnames.sort()
+
+    ide_list = ["iar", "uvision5"]
```

Comment:
seems like a list of tests and ide should be in a json file for easy extension. Maybe another pr though